### PR TITLE
Feature: not just utc time but also support other timezone by setting in prefs page

### PR DESCRIPTION
**Pull request by ChasingZenith**

### CODE REVIEW

**Comment by swsnr** [comment]
I'm not sure I'd like to merge this.  I feel like it's out of scope for this extension, and I have no use case for this feature.

File: src/extension.ts [hunk 1]
```diff
@@ -94,8 +105,8 @@ class EnabledExtension {
    * for this extension.
    */
   updateClockDisplay() {
-    const now = GLib.DateTime.new_now_utc();
-    const utcNow = now.format(this.settings.get_string("clock-format"));
+    const now = GLib.DateTime.new_now_utc().add_hours(this.settings.get_int("utc-offset"));
```
Comment (swsnr) [inline src/extension.ts:108]: That's definitely not the right way to handle timezone offsets; in particular format codes such as `%z` will be entirely unaware if you apply the offset this way.

You need to create a date time object with proper offset information attached.

File: src/prefs.ts [hunk 2]
```diff
@@ -97,6 +97,22 @@ export default class UTCClockPreferences extends ExtensionPreferences {
       "text",
       Gio.SettingsBindFlags.DEFAULT,
     );
+    const utcOffsetSpinRow = new Adw.SpinRow ({
+      title: "Time Zone Offset",
+      adjustment: new Gtk.Adjustment({
+          lower: -11,
+          upper: 14,
```
Comment (swsnr) [inline src/prefs.ts:104]: Where do these limits come from?

File: schemas/org.gnome.shell.extensions.swsnr-utc-clock.gschema.xml [hunk 3]
```diff
@@ -6,5 +6,10 @@
       <summary>The format to use to display the UTC time</summary>
       <description>A GLib DateTime format string which describes the format to use for the clock</description>
     </key>
+    <key name="utc-offset" type="i">
+      <default>0</default>
+      <summary>offset (in hours) relative to utc+0</summary>
+      <description>A int32 which describes the offet of the time zone in hours</description>
```
Comment (swsnr) [inline schemas/org.gnome.shell.extensions.swsnr-utc-clock.gschema.xml:12]: The type information is entirely redundant here; just write  "A valid offset from UTC" as summary, and omit the description.

**Review by swsnr — CHANGES_REQUESTED** [review]
Not promising to merge it even though, but for me to even consider it, please address the following points:

- Address all remarks in this review.
- Fix the error on Github actions.
- Rebase onto master and move the new settings UI into the blueprint template.

**Comment by ChasingZenith** [comment]
Thank you for your kindly review! What you point out is completely helpful. Sorry for the bad quality since it is my first touch to gtk, gnome extension, so I am not very familiar with it. I will try to solve this some days later, since I am quite busy this week.

**Comment by swsnr** [comment]
@ChasingZenith Do you still intend to work on this?  If I don't hear anything from you, I'll close this in a few days.

**Comment by ChasingZenith** [comment]
Sorry, I'll work on this some times later. Please close it for the moment.

**Pull request closed without merging**